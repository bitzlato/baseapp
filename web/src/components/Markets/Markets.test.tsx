import { shallow, ShallowWrapper } from 'enzyme';
import { TestComponentWrapper } from 'src/lib/test';
import { Markets, MarketsProps } from '.';

const data = [
  ['ETH/BTC', '0.223100', '+50.00%'],
  ['ETH/LTC', '0.223100', '+25.00%'],
  ['LTC/BTC', '0.223100', '-5.00%'],
];
const onSelect = jest.fn();

const defaultProps: MarketsProps = {
  data,
  onSelect,
};

const setup = (props?: Partial<MarketsProps>) =>
  shallow(
    <TestComponentWrapper>
      <Markets {...{ ...defaultProps, ...props }} />
    </TestComponentWrapper>,
  );

describe('Markets', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('should render', () => {
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should render empty data', () => {
    wrapper = setup({ data: [] });
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should set selected market in props', () => {
    const keyIndex = 0;
    const selectedKey = 'ETH/LTC';

    const component = setup({
      data,
      rowKeyIndex: keyIndex,
      selectedKey,
      onSelect,
    }).render();

    const selectedRow = component.find('.cr-table__row--selected').first().get(0)?.tagName;
    expect(selectedRow).toBe('tr');
  });
});
