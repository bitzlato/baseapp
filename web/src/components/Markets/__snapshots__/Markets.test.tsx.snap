// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markets should render 1`] = `
<div
  class="cr-markets"
>
  <div
    class="cr-table-container"
  >
    <div
      class="cr-table-header__content"
    >
      <div
        class="cr-title-component"
      >
        Markets
      </div>
    </div>
    <table
      class="cr-table"
    >
      <thead
        class="cr-table__head"
      >
        <tr
          class="cr-table__head-row"
        >
          <th>
            Pair
          </th>
          <th>
            Price
          </th>
          <th>
            24h Change
          </th>
        </tr>
      </thead>
      <tbody
        class="cr-table__body"
      >
        <tr
          class=""
        >
          <td>
            <p
              class="reset_base__1sfgjt80 sprinkles_color_text_default__w76aifww sprinkles_whiteSpace_nowrap__w76aif1m1 Text_text__1qy84op6 Text__1qy84op0 sprinkles_fontFamily_brand__w76aif1ll Text_text_variant_caption__1qy84oph Text__1qy84op5 sprinkles_fontSize_caption_mobile__w76aifvt Text_text_gutterBottom_false__1qy84opj "
            >
              <span>
                ETH
              </span>
              /
              <span>
                BTC
              </span>
            </p>
          </td>
          <td>
            <p
              class="reset_base__1sfgjt80 sprinkles_color_text_default__w76aifww sprinkles_textOverflow_ellipsis__w76aif1m2 Text_text__1qy84op6 Text__1qy84op0 sprinkles_fontFamily_brand__w76aif1ll Text_text_variant_caption__1qy84oph Text__1qy84op5 sprinkles_fontSize_caption_mobile__w76aifvt Text_text_gutterBottom_false__1qy84opj "
              title="0.223100"
            >
              0.223100
            </p>
          </td>
          <td>
            <span
              class="__positive"
            >
              +50.00%
            </span>
          </td>
        </tr>
        <tr
          class=""
        >
          <td>
            <p
              class="reset_base__1sfgjt80 sprinkles_color_text_default__w76aifww sprinkles_whiteSpace_nowrap__w76aif1m1 Text_text__1qy84op6 Text__1qy84op0 sprinkles_fontFamily_brand__w76aif1ll Text_text_variant_caption__1qy84oph Text__1qy84op5 sprinkles_fontSize_caption_mobile__w76aifvt Text_text_gutterBottom_false__1qy84opj "
            >
              <span>
                ETH
              </span>
              /
              <span>
                LTC
              </span>
            </p>
          </td>
          <td>
            <p
              class="reset_base__1sfgjt80 sprinkles_color_text_default__w76aifww sprinkles_textOverflow_ellipsis__w76aif1m2 Text_text__1qy84op6 Text__1qy84op0 sprinkles_fontFamily_brand__w76aif1ll Text_text_variant_caption__1qy84oph Text__1qy84op5 sprinkles_fontSize_caption_mobile__w76aifvt Text_text_gutterBottom_false__1qy84opj "
              title="0.223100"
            >
              0.223100
            </p>
          </td>
          <td>
            <span
              class="__positive"
            >
              +25.00%
            </span>
          </td>
        </tr>
        <tr
          class=""
        >
          <td>
            <p
              class="reset_base__1sfgjt80 sprinkles_color_text_default__w76aifww sprinkles_whiteSpace_nowrap__w76aif1m1 Text_text__1qy84op6 Text__1qy84op0 sprinkles_fontFamily_brand__w76aif1ll Text_text_variant_caption__1qy84oph Text__1qy84op5 sprinkles_fontSize_caption_mobile__w76aifvt Text_text_gutterBottom_false__1qy84opj "
            >
              <span>
                LTC
              </span>
              /
              <span>
                BTC
              </span>
            </p>
          </td>
          <td>
            <p
              class="reset_base__1sfgjt80 sprinkles_color_text_default__w76aifww sprinkles_textOverflow_ellipsis__w76aif1m2 Text_text__1qy84op6 Text__1qy84op0 sprinkles_fontFamily_brand__w76aif1ll Text_text_variant_caption__1qy84oph Text__1qy84op5 sprinkles_fontSize_caption_mobile__w76aifvt Text_text_gutterBottom_false__1qy84opj "
              title="0.223100"
            >
              0.223100
            </p>
          </td>
          <td>
            <span
              class="__negative"
            >
              -5.00%
            </span>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="cr-table-background"
    />
  </div>
  <div
    class="cr-search"
  >
    <span
      class="cr-search__icon"
    >
      <img
        alt="search icon"
        src="test-file-stub"
      />
    </span>
    <input
      class="cr-search__input"
      placeholder="Search"
      type="text"
      value=""
    />
    <span
      class="cr-search__cancel"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.5 0C11.6475 0 15 3.3525 15 7.5C15 11.6475 11.6475 15 7.5 15C3.3525 15 0 11.6475 0 7.5C0 3.3525 3.3525 0 7.5 0ZM10.1925 3.75L7.5 6.4425L4.8075 3.75L3.75 4.8075L6.4425 7.5L3.75 10.1925L4.8075 11.25L7.5 8.5575L10.1925 11.25L11.25 10.1925L8.5575 7.5L11.25 4.8075L10.1925 3.75Z"
          fill="var(--icons)"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`Markets should render empty data 1`] = `
<div
  class="cr-markets"
>
  <div
    class="cr-table-container"
  >
    <div
      class="cr-table-header__content"
    >
      <div
        class="cr-title-component"
      >
        Markets
      </div>
    </div>
    <table
      class="cr-table"
    >
      <thead
        class="cr-table__head"
      >
        <tr
          class="cr-table__head-row"
        >
          <th>
            Pair
          </th>
          <th>
            Price
          </th>
          <th>
            24h Change
          </th>
        </tr>
      </thead>
      <tbody
        class="cr-table__body"
      />
    </table>
    <div
      class="cr-table-background"
    />
  </div>
  <div
    class="cr-search"
  >
    <span
      class="cr-search__icon"
    >
      <img
        alt="search icon"
        src="test-file-stub"
      />
    </span>
    <input
      class="cr-search__input"
      placeholder="Search"
      type="text"
      value=""
    />
    <span
      class="cr-search__cancel"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.5 0C11.6475 0 15 3.3525 15 7.5C15 11.6475 11.6475 15 7.5 15C3.3525 15 0 11.6475 0 7.5C0 3.3525 3.3525 0 7.5 0ZM10.1925 3.75L7.5 6.4425L4.8075 3.75L3.75 4.8075L6.4425 7.5L3.75 10.1925L4.8075 11.25L7.5 8.5575L10.1925 11.25L11.25 10.1925L8.5575 7.5L11.25 4.8075L10.1925 3.75Z"
          fill="var(--icons)"
        />
      </svg>
    </span>
  </div>
</div>
`;
